libraries/app/database_api.cpp

```diff
@@ -624,6 +625,87 @@ vector<optional<account_object>> database_api_impl::get_accounts(const vector<ac
    return result;
 }
 
+vector<limit_order_object> database_api::get_account_limit_orders( const string& name_or_id, const string &base, const string &quote, uint32_t limit, optional<limit_order_id_type> ostart_id, optional<price> ostart_price)
+{
+   return my->get_account_limit_orders( name_or_id, base, quote, limit, ostart_id, ostart_price );
+}
+
+vector<limit_order_object> database_api_impl::get_account_limit_orders( const string& name_or_id, const string &base, const string &quote, uint32_t limit, optional<limit_order_id_type> ostart_id, optional<price> ostart_price)
+{
+   FC_ASSERT( limit <= 101 );
+
+   vector<limit_order_object>   results;
+   const account_object*        account = nullptr;
+   uint32_t                     count = 0;
+
+   if (std::isdigit(name_or_id[0]))
+      account = _db.find(fc::variant(name_or_id, 1).as<account_id_type>(1));
+   else
+   {
+       const auto& idx = _db.get_index_type<account_index>().indices().get<by_name>();
+       auto itr = idx.find(name_or_id);
+       if (itr != idx.end())
+           account = &*itr;
+   }
+   if (account == nullptr)
+       return results;
+
+   auto assets = lookup_asset_symbols( {base, quote} );
+   FC_ASSERT( assets[0], "Invalid base asset symbol: ${s}", ("s",base) );
+   FC_ASSERT( assets[1], "Invalid quote asset symbol: ${s}", ("s",quote) );
+
+   auto base_id = assets[0]->id;
+   auto quote_id = assets[1]->id;
+
+   if (ostart_price.valid()) {
+       FC_ASSERT(ostart_price->base.asset_id == base_id, "Base asset inconsistent with start price");
+       FC_ASSERT(ostart_price->quote.asset_id == quote_id, "Quote asset inconsistent with start price");
+   }
+
+   const auto& index_by_account = _db.get_index_type<limit_order_index>().indices().get<by_account>();
+   limit_order_multi_index_type::index<by_account>::type::const_iterator lower_itr;
+   limit_order_multi_index_type::index<by_account>::type::const_iterator upper_itr;
```

Comment:
Actually `upper_itr` can be initialized here, no need to be initialized in `if/else` block below, since the results will be the same.

Reply:
has initialize upper_itr in each of if/else if/else branch, is it ok?